### PR TITLE
Menu Sequence

### PR DESCRIPTION
**Pull request by AlekseyPanas**

### CODE REVIEW

File: Assets/Prefabs/MenuSequence/AWaitFor.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+using UnityEngine;
+
+public delegate void FinishedTask();
+
+/** Object will send an event when they finish a task. An object can hold a bunch of IWaitFor instances and wait for them all to finish */
+public abstract class AWaitFor: MonoBehaviour {
+    public event FinishedTask FinishedTaskEvent = delegate {}; protected void invokeFinishedTask() { FinishedTaskEvent(); }
```
Comment (1whatleytay) [inline Assets/Prefabs/MenuSequence/AWaitFor.cs:7]: One line per statement please!

File: Assets/Prefabs/MenuSequence/MenuBgScene/CampfireLightFlicker.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    private float baseIntensity;
+
+    // Start is called before the first frame update
+    void Start() {
+        baseIntensity = GetComponent<Light>().intensity;
+    }
+
+    // Update is called once per frame
+    void Update() {
+        Light light = GetComponent<Light>();
```
Comment (1whatleytay) [inline Assets/Prefabs/MenuSequence/MenuBgScene/CampfireLightFlicker.cs:18]: Do not use GetComponent in update if possible, it's known to be pretty expensive. Make a field and store it there in `Start` if you can.

File: Assets/Prefabs/MenuSequence/AWaitFor.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+using UnityEngine;
+
+public delegate void FinishedTask();
+
+/** Object will send an event when they finish a task. An object can hold a bunch of IWaitFor instances and wait for them all to finish */
+public abstract class AWaitFor: MonoBehaviour {
```
Comment (1whatleytay) [inline Assets/Prefabs/MenuSequence/AWaitFor.cs:6]: Why is AWaitFor abstract? There's nothing for the child to implement as far as I can see.

File: Assets/Prefabs/MenuSequence/MenuBgScene/Visuals/TitleFrames/15.png
Comment (1whatleytay) [inline Assets/Prefabs/MenuSequence/MenuBgScene/Visuals/TitleFrames/15.png:1]: 😐

File: Assets/Prefabs/RespawnPoint/RespawnPoint.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
```
Comment (1whatleytay) [inline Assets/Prefabs/RespawnPoint/RespawnPoint.cs:21]: Seems like a data class. Maybe drop the methods?

File: Assets/Prefabs/MenuSequence/AWaitFor.cs (hunk 1 above)
Comment (1whatleytay) [reply Assets/Prefabs/MenuSequence/AWaitFor.cs:7]: Or one statement per line.

File: Assets/Prefabs/MenuSequence/AWaitFor.cs (hunk 3 above)
Comment (AlekseyPanas) [reply Assets/Prefabs/MenuSequence/AWaitFor.cs:6]: So that it can be injected into Unity in various places

Comment (1whatleytay) [reply Assets/Prefabs/MenuSequence/AWaitFor.cs:6]: If it's not abstract does unity not recognize it

Comment (AlekseyPanas) [reply Assets/Prefabs/MenuSequence/AWaitFor.cs:6]: Unity only allows fields that inherit MonoBehavior

Comment (AlekseyPanas) [reply Assets/Prefabs/MenuSequence/AWaitFor.cs:6]: Ill see if theres a way to get around it

File: Assets/Prefabs/MenuSequence/MenuBgScene/Visuals/TitleFrames/15.png
Comment (AlekseyPanas) [reply Assets/Prefabs/MenuSequence/MenuBgScene/Visuals/TitleFrames/15.png:1]: What? lol

File: Assets/Prefabs/MenuSequence/AWaitFor.cs (hunk 3 above)
Comment (AlekseyPanas) [reply Assets/Prefabs/MenuSequence/AWaitFor.cs:6]: Tried but found no way to solve this. There are many posts on the Unity Forums complaining about this issue, here is one you can skim: https://forum.unity.com/threads/serialized-interface-fields.1238785/ 
If we really need to use interfaces with injection, I think we'll label it as GameObject and use a GetComponent check to test that the interface exists on the injected objects (and maybe send a warning if not)

Comment (1whatleytay) [reply Assets/Prefabs/MenuSequence/AWaitFor.cs:6]: That's so interesting. Thanks for looking into it.

**Review by 1whatleytay — DISMISSED** [review]

**Comment by 1whatleytay** [comment]
Sure!

File: Assets/Prefabs/FadeToBlackPanel/FadeToBlackPanel.cs
Comment (PyroPoro) [inline Assets/Prefabs/FadeToBlackPanel/FadeToBlackPanel.cs:1]: same thing that taylor said, can move col = getcomponent<image> out of the while loop to save runtime

File: Assets/Prefabs/MenuSequence/Cutscene/CutsceneData.cs
Comment (PyroPoro) [inline Assets/Prefabs/MenuSequence/Cutscene/CutsceneData.cs:1]: would prefer if the tooltip is on a line above the attribute declaration

File: Assets/Prefabs/FadeToBlackPanel/FadeToBlackPanel.cs
Comment (PyroPoro) [reply Assets/Prefabs/FadeToBlackPanel/FadeToBlackPanel.cs:1]: ok idk why this didnt mark the specific file

**Review by 1whatleytay — APPROVED** [review]
Don't have time to look over. Just approving to unblock you.

**Review by PyroPoro — APPROVED** [review]
changes look good

**Pull request merged**